spark/src/main/java/org/apache/iceberg/spark/source/RowDataReader.java

```diff
@@ -252,4 +258,29 @@ private static Object convertConstant(Type type, Object value) {
     }
     return value;
   }
+
+  private OrcRowFilter getOrcRowFilter(FileScanTask task) {
+    FileScanTask fileScanTask;
+    if (task instanceof BaseFileScanTask.SplitScanTask) {
+      fileScanTask = ((BaseFileScanTask.SplitScanTask) task).underlyingFileScanTask();
+    } else {
+      fileScanTask = task;
+    }
+    if (fileScanTask instanceof OrcRowFilterableFileScanTask) {
+      return ((OrcRowFilterableFileScanTask) fileScanTask).orcRowFilter();
+    }
+    return null;
+  }
+
+  private void validateRowFilterRequirements(FileScanTask task, OrcRowFilter filter) {
+    Preconditions.checkArgument(task.file().format() == FileFormat.ORC, "Row filter can only be applied to ORC files");
+    Preconditions.checkArgument(task.spec().fields().size() == 0,
+        "Row filter can only be applied to unpartitioned tables");
```

Comment:
I don't follow this requirement

Reply:
We don't handle it for unpartitioned tables today, else I will need to do special handling in RowDataReader. Since Opal tables are not unpartitioned, it isn't worth the effort.

Reply:
Actually you are correct. It was a limitation but I changed way I handle row level filtering and did not revisit this requirement. I will revisit this and remove if required.

Reply:
I revisited this, the current process won't handle row filters on identity partitioned columns which come from metadata rather than data, since the values of these columns are not available while reading the data in OrcIterable. Since opal tables are not partitioned, I prefer keeping this restriction for now and revisit later if required.

Reply:
makes sense.